packages/rpc-core/src/rpc-methods/getTransaction.ts

```diff
@@ -0,0 +1,297 @@
+    data: Base64EncodedDataResponse;
+};
+
+type TransactionMetaBase = Readonly<{
+    /** Error if transaction failed, null if transaction succeeded. */
+    err: TransactionError | null;
+    /** fee this transaction was charged */
+    fee: LamportsUnsafeBeyond2Pow53Minus1;
+    /** array of account balances from before the transaction was processed */
+    preBalances: readonly LamportsUnsafeBeyond2Pow53Minus1[];
+    /** array of account balances after the transaction was processed */
+    postBalances: readonly LamportsUnsafeBeyond2Pow53Minus1[];
+    /** List of token balances from before the transaction was processed or omitted if token balance recording was not yet enabled during this transaction */
+    preTokenBalances?: readonly TokenBalance[];
+    /** List of token balances from after the transaction was processed or omitted if token balance recording was not yet enabled during this transaction */
+    postTokenBalances?: readonly TokenBalance[];
+    /** array of string log messages or null if log message recording was not enabled during this transaction */
+    logMessages: readonly string[] | null;
+    /** transaction-level rewards */
+    rewards: readonly TransactionReward[] | null;
+    /**
+     * Transaction status
+     * @deprecated
+     */
+    status: TransactionStatus;
+    /** the most-recent return data generated by an instruction in the transaction */
+    returnData?: ReturnData;
+    /** number of compute units consumed by the transaction */
+    computeUnitsConsumed?: U64UnsafeBeyond2Pow53Minus1;
+}>;
+
+type AddressTableLookup = Readonly<{
+    /** public key for an address lookup table account. */
+    accountKey: Base58EncodedAddress;
+    /** List of indices used to load addresses of writable accounts from a lookup table. */
+    writableIndexes: readonly number[];
+    /** List of indices used to load addresses of readonly accounts from a lookup table. */
+    readableIndexes: readonly number[];
+}>;
+
+type TransactionBase = Readonly<{
+    message: {
+        recentBlockhash: Blockhash;
+    };
+    signatures: readonly Base58EncodedBytes[];
+}>;
+
+type TransactionInstruction = Readonly<{
+    programIdIndex: number;
+    accounts: readonly number[];
+    data: Base58EncodedBytes;
+}>;
+
+type TransactionJson = TransactionBase &
+    Readonly<{
+        message: {
+            accountKeys: readonly Base58EncodedAddress[];
+            header: {
+                numReadonlySignedAccounts: number;
+                numReadonlyUnsignedAccounts: number;
+                numRequiredSignatures: number;
+            };
+            instructions: readonly TransactionInstruction[];
+        };
+    }>;
+
+type PartiallyDecodedTransactionInstruction = Readonly<{
+    accounts: readonly Base58EncodedAddress[];
+    data: Base58EncodedBytes;
+    programId: Base58EncodedAddress;
+}>;
+
+type ParsedTransactionInstruction = Readonly<{
+    parsed: {
+        type: string;
+        info: object;
```

Comment:
I think this can be undefined.

https://github.com/solana-labs/solana/blob/32072da36bdf645b6f09e9e1f453fbfe4db2f3e9/transaction-status/src/parse_instruction.rs#L89-L94

Looks like `info` has `#[serde(default, skip_serializing_if = "Value::is_null")]` on it.

Reply:
Yup, there are parsed instructions that use `Value::default()` and that's `null`.

https://github.com/solana-labs/solana/blob/32072da36bdf645b6f09e9e1f453fbfe4db2f3e9/transaction-status/src/parse_stake.rs#L272-L275

Will fix!